### DISCUSSION TITLE
[2023-12-18] Team GUI Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#8572] [Weekly Demo Day]
@farmaazon:

* #8535
  * Removed system cursor
  * Changed shortcut for Code Editor
  * Added `import Standard.Visualization` as a workaround allowing visualization previews. See https://github.com/orgs/enso-org/discussions/6832#discussioncomment-7789804

https://github.com/enso-org/enso/assets/3919101/38d822ac-28df-4a9c-a2dc-dee7adbd58ea

* #8559 - includes a fix for a bug spotted by one of them.

@vitvakatu:

* #8531

https://github.com/enso-org/enso/assets/6566674/052462ca-1a09-428f-ba44-437355e2da1c

* #8568 - still in progress, taken by @farmaazon for this week

**Comment by somebody1234** [comment]
- Debug & fix crash on `text/openFile`: #8514
- Properly integrate GUI2 with dashboard (cloud auth): #8542

**Comment by MichaelMauderer** [comment]
Worked on 
* #8525
* #8524

[Screencast from 18-12-23 14:40:29.webm](https://github.com/enso-org/enso/assets/1428930/a6648869-6017-46b7-a606-2ca4505a0a48)

**Comment by mwu-tow** [comment]
* Today's nightly build failure is being investigated.
* Installer development is progressing nicely, tonight I expect to have the first build with the new installer for testing.

**Comment by kazcw** [comment]
Finished #8367 [in code review now].
- The module is edited by changing the AST, not the text representation.
- `IdMap`s no longer need to be maintained in parallel with the module content; they are snapshots produced as needed from the ASTs.

**Comment by Frizi** [comment]
- Implemented all cases of argument assignment and clearing https://github.com/enso-org/enso/pull/8545
- Now reviewing AST edit branch, planning to help porting existing code to use new editing APIs.

https://github.com/enso-org/enso/assets/919491/513b823b-eb2c-45d8-88ac-4971ba061c59

**Comment by hubertp** [comment]
Relevant bug fixes in backend:
- Fix serialization of alias analysis graph #8550 
- Stop parsing METADATA section for IdMap #8574
